contracts/mockContracts/MockDGXDex.sol

```diff
@@ -0,0 +1,187 @@
+pragma solidity ^0.4.18;
+
+//ask:
+//dst address change?
+//should init another token outside in test?
+
+// blocknumber + 5 >= current block number ? revert
+// signatures - skip for now.
+
+import "../ERC20Interface.sol";
+import "../Withdrawable.sol";
+import "./ecverify.sol";
+
+
+/// @title Mock Digix DEX
+/// @author Tal Baneth
+/// @dev a dummy contract that simulates Digix contract for purchase/sell with verified signatures. 
+contract MockDGXDEX is Withdrawable {
+    uint  constant internal G_TO_MG = (10**3);
+    uint  constant internal DGX_DECIMALS = (10**9);
+    
+    ERC20 token;
+    address feedSigner;
+    bool public sigCheck = true;
+    bool public blockCheck = true;
+
+    function MockDGXDEX(ERC20 _token, address _feedSigner, address _admin) public {
+        admin = _admin;
+        token = _token;
+        feedSigner = _feedSigner;
+    }
+    
+    function () public payable {}
+    
+    event Purchase(bool success, uint256 purchasedAmount);
+    event Sell(bool success, uint256 amountWei);
+
+    function setSigCheck (bool _sigCheck) public onlyAdmin {
+        sigCheck = _sigCheck;
+    }
+
+    function setBlockCheck (bool _blockCheck) public onlyAdmin {
+        blockCheck = _blockCheck;
+    }
+
+    function purchase(uint256 block_number,
+                      uint256 nonce,
+                      uint256 weiPerDgxMg,
+                      address signer,
+                      bytes signature) payable public returns (bool success, uint256 purchasedAmount) {
+
+        uint256 tokenAmount;
+
+        if( sigCheck) {
+            bool verified;
+            address actual_signer;
+            
+            (verified, actual_signer) =   
+            verify_signed_price(block_number,
+                                nonce,
+                                weiPerDgxMg,
+                                signer,
+                                signature);
+
+            require(verified);
+        }
+        if(blockCheck) {
+            require((block_number + 5) >= block.number);
```

Comment:
this value - 5 - better be configurable...
anyway would rather not use numbers but  const values.d

Reply:
Done. Added a const.